packages/expo-file-system/ios/EXFileSystem/EXFileSystem.m

```diff
@@ -866,4 +921,43 @@ - (NSNumber *)freeDiskStorage {
   return nil;
 }
 
+#pragma mark - NSURLSessionDelegate
+
+- (void)URLSession:(NSURLSession *)session downloadTask:(NSURLSessionDownloadTask *)downloadTask didFinishDownloadingToURL:(NSURL *)location
+{
+  EXSessionTaskDelegate *exTask = _tasks[downloadTask];
+  if (exTask) {
+    [exTask URLSession:session downloadTask:downloadTask didFinishDownloadingToURL:location];
+    [_tasks removeObjectForKey:downloadTask];
+    [self _unregisterIfResumableTaskDelegate:exTask];
+  }
+}
+
+- (void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)task didCompleteWithError:(NSError *)error
+{
+  EXSessionTaskDelegate *exTask = _tasks[task];
+  if (exTask) {
+    [exTask URLSession:session task:task didCompleteWithError:error];
+    [_tasks removeObjectForKey:task];
+    [self _unregisterIfResumableTaskDelegate:exTask];
+  }
+}
+
+- (void)URLSession:(NSURLSession *)session downloadTask:(NSURLSessionDownloadTask *)downloadTask
+                                           didWriteData:(int64_t)bytesWritten
+                                      totalBytesWritten:(int64_t)totalBytesWritten
+                              totalBytesExpectedToWrite:(int64_t)totalBytesExpectedToWrite
+{
+  if (bytesWritten > 0) {
```

Comment:
Should this logic be here or in the implementation of the `TaskDelegate`? I feel if `EXFileSystem` is supposed to be a dispatcher, we shouldn't omit any of the events to keep the data flow simpler.